src/prefect/results.py

```diff
@@ -143,6 +143,38 @@ def get_default_persist_setting() -> bool:
     return PREFECT_RESULTS_PERSIST_BY_DEFAULT.value()
 
 
+def flow_features_require_result_persistence(flow: "Flow") -> bool:
```

Comment:
I don't think we should ever persist results if a user didn't request it

Reply:
Or said differently, if persistence is set to off, we shouldn't ever override that